### ISSUE TITLE
Repeated class names are removed (Blaze)

### ISSUE THREAD

**Issue by dalgard** [meteor/meteor#3628]
The `class` attribute or `className` property is a string and may sometimes need to be dealt with as a string, without alteration. For instance, in CSS, a selector may look like this `[class*='big big big header]`.

Silly as it may seem, this is actually used in Semantic UI to create a responsive grid (`four wide column eight wide tablet` – `wide` is repeated), but it seems that Meteor, or possibly jQuery/Blaze, is "cleaning up" the classes, so that repeated classes are removed.

Any chance of changing this behavior?

**Comment by dgreensp** [comment]
Hi @dalgard, wow, that is very surprising to me.  I expect Semantic UI will run into a lot of problems with that approach, in interactions with UI frameworks and even future DOM APIs.  I would go so far as to say it runs counter to the semantics of the HTML5 `class` attribute, in which the set of space-separated tokens is supposed to "represent the various classes that the element belongs to" (https://html.spec.whatwg.org/multipage/dom.html#classes).

That's not to say we would never implement the feature of preserving the class attribute.  I would accept a PR if it doesn't add too much complexity.

**Comment by msurdi** [comment]
I'm also facing this problem. Have you found any workaround?

**Comment by dalgard** [comment]
I'm using a block helper which applies the classes in the keyword argument `classes` to `template_instance.firstNode` after render. It works okay.

**Comment by ansarizafar** [comment]
Could you please share the code of block helper.

**Comment by dalgard** [comment]
Updated my answer.

**Comment by msurdi** [comment]
Ok, I'm finally using this workaround until this issue is fixed: 

```javascript
Template.layout.onRendered(function () {
  let elements = this.findAll("[data-class]") || [];
  for (let element of elements) {
    element.setAttribute("class", element.getAttribute("data-class"));
  }
});
````

Then, when it happens that I need to preserve the class ordering and/or duplicates, instead of using a `class` attribute as I would normally do, I rename it to `data-class`.

Do you think this could have any undesired side effect? (I literally started last week using meteor).

Thanks

**Comment by dalgard** [comment]
In many cases it won't work, unfortunately, since element and templates may be added and removed from the page by things like `#each` and `#if` helpers, without the layout re-rendering.

**Comment by msurdi** [comment]
Yes, I've just found it out that the hard way :-(. Looking for a better solution now.

**Comment by msurdi** [comment]
Ok, this seems to be working better, although probably a lot less performant (not an issue for me right now):


```javascript
Template.body.onRendered(function () {
  $("body").bind("DOMSubtreeModified", function () {
    let elements = $("[data-class]");
    elements.each(function (index, element) {
      element.setAttribute("class", element.getAttribute("data-class"));
    });
  });
});
```

**Comment by dalgard** [comment]
I'm sure you are aware, that this is the kind of code that becomes legacy in a project and then suddenly blows up unexplainably ;)

I'd go with the block helper.

**Comment by msurdi** [comment]
On the other hand, as soon as this bug is fixed, the snippet can be removed and everything else would look like this never happened. So for me following this issue and adding a link to it in the source is a good  tradeoff. Anyway, this is just a hobby project of myself, so not a big concern at all, I'll probably forget about it in a few weeks. Thanks for your input.

**Comment by dalgard** [comment]
That's totally fair. Don't expect this bug to be fixed before we move on to a completely new view library, though. Maybe it will be different in Blaze 2.0, but maybe it will die with Blaze itself.

**Comment by timothyarmes** [comment]
I'm adding my vote to get this fixed. You have official semantic UI support yet it's not possible to create a responsive layout without hacky workarounds.

**Comment by talha-asad** [comment]
This is being discussed at #5632. Lets close this and get this patched in a PR.

**Comment by hwillson** [comment]
This issue is closely related to issue #5632, which was closed after being migrated to meteor/blaze#30. Blaze issue meteor/blaze#30 was then closed as it was decided Blaze is behaving properly based on the HTML5 spec (see this [comment](https://github.com/meteor/blaze/issues/30#issuecomment-218541443)). Given the decision made in meteor/blaze#30, this issue can also be closed (since the removal of repeated class names follows the HTML5 spec). Thanks!

**Comment by abernix** [comment]
Thanks, @hwillson. :)
